### DISCUSSION TITLE
[nextjs@13] what is the precedence attribute in head?

### DISCUSSION THREAD

**Discussion by yceffort** [vercel/next.js#44372] [Help] [answered]
https://beta.nextjs.org/docs/api-reference/file-conventions/head#supported-head-tags

I have a question in the beta document. 
what is the `precedence` attribute in head exactly? 
Is this new React-specific attribute for link tag? how this attribute works?

Is there a specific document describing this attribute? I didn't find this attribute in the React repository either.

**Comment by icyJoseph** [comment, marked as answer]
It is a `react-dom` attribute, so it's not `react`, nor `next.js`. 

I think what it does is, label links ,with stylesheet rel, as they appear in the DOM, and honors that order of appearance, even if the resource is found multiple times. 

It is true that it is not documented very well at the moment, but I did find it in the `react-dom` repository:

- https://github.com/facebook/react/blob/8e2bde6f2751aa6335f3cef488c05c3ea08e074a/packages/react-dom-bindings/src/server/fizz-instruction-set/ReactDOMFizzInstructionSet.js#L55
- https://github.com/facebook/react/blob/5dfc485f6957bf941fcd9565e0271aea019d7c9e/packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js#L1284
- https://github.com/facebook/react/blob/bfcbf3306794eb3de37d866551127fc2bdae53d7/packages/react-dom-bindings/src/client/ReactDOMFloatClient.js#L351
- https://github.com/facebook/react/blob/bfcbf3306794eb3de37d866551127fc2bdae53d7/packages/react-dom-bindings/src/server/ReactDOMFloatServer.js#L489

I'd say that, while it is not properly documented, other than through one of its use cases, and perhaps the only use case, one should not lose sleep over it, nor go about writing articles and such trying to get a hot take. I am saying this in general, not just for you, who might be genuinely curious and interested on the topic.

**Reply by yceffort** [reply]
thanks for your help. 

This is the only hint I got while looking for the nextjs repository.

https://github.com/vercel/next.js/blob/d3c292dd4abba9f91ddf2feff5f9ad935e52c052/packages/next/server/app-render.tsx#L1123-L1124

I asked because I didn't know exactly where to find it and get information about it. Maybe it's because I'm not good at reading source code, but it's not that I haven't tried it.

It is also true that the official document lacks explanation, so I hope it will be added later.

Anyway, thank you for your help.
